src/catchup/StateConsistencyWork.cpp

```diff
@@ -0,0 +1,90 @@
+#include "catchup/StateConsistencyWork.h"
+#include "herder/Herder.h"
+#include "history/FileTransferInfo.h"
+#include "history/HistoryManager.h"
+#include "historywork/GetAndUnzipRemoteFileWork.h"
+
+namespace stellar
+{
+
+EnsureHerderConsistencyWork::EnsureHerderConsistencyWork(Application& app,
+                                                         uint32_t ledgerSeq,
+                                                         TmpDir const& tmpDir)
+    : Work(app, "herder-state-consistency", BasicWork::RETRY_NEVER)
+    , mLedger(ledgerSeq)
+    , mDownloadDir(tmpDir)
+{
+}
+
+void
+EnsureHerderConsistencyWork::doReset()
+{
+    mDownloadHeader.reset();
+}
+
+BasicWork::State
+EnsureHerderConsistencyWork::doWork()
+{
+    // No-op if Herder is already up to date
+    if (mApp.getHerder().trackingConsensusLedgerIndex() >= mLedger)
+    {
+        return State::WORK_SUCCESS;
+    }
+
+    auto checkpoint =
```

Comment:
this doesn't seem right: I think we already downloaded and verified the last ledger in the range with `VerifyLedgerChainWork` (that also checks that it matches the trusted anchor if set), this new Work is downloading a potentially different ledger header that is not trusted (also, it doesn't deal with malicious checkpoint files at all).

Reply:
you're right: this work doesn't need to download anything, we just need to reuse the file that's already verified (not sure why I left it there...). I updated it now.